libs/libc/stdio/lib_vasprintf.c

```diff
@@ -119,14 +119,18 @@ int vasprintf(FAR char **ptr, FAR const IPTR char *fmt, va_list ap)
 #endif
 
   /* Return a pointer to the string to the caller.  NOTE: the memstream put()
-   * method has already added the NUL terminator to the end of the string
+   * method has already added the NULL terminator to the end of the string
```

Comment:
First time I ever see it written like that... I thought it was a typo.

Anyway, fixed.

Reply:
Here is some discussion https://news.ycombinator.com/item?id=1401327
However I just searched on the Internet and saw that most of the articles state "null-terminated", so maybe I was learning C from very old books some time ago that still used NUL term for string termination. Seems like your change was also correct.

Reply:
Also some more info here https://www.ibm.com/docs/en/i/7.3?topic=essiccatus-nulls-nuls-in-c-c-applications-that-use-sql

Reply:
Oldtimers remember TTY really being a Teletype terminal. Here is where `NUL` is from.
 
![image](https://user-images.githubusercontent.com/1945821/228224038-e2a4ce1d-f8be-4a24-9540-d9ddc7998f67.png)

Reply:
Hmm indeed both seem "correct", although the NULL version is much more widely adopted. I have never seen it before as NUL (or maybe I also considered a typo?).

I propose to leave it as is for the moment. If we want to rectify this, then it can be a dedicated PR.
I see that there are 163 total occurrences in the code base. So, if it is to be changed, then it should be done for all of them at the same time, for consistency.